src/base-client.ts

```diff
@@ -72,6 +73,83 @@ export class BaseSlackAPIClient implements BaseSlackClient {
     return await this.createBaseResponse(response);
   }
 
+  async fileUploadV2(
+    args: FileUploadV2Args,
+  ) {
+    const { file_uploads } = args;
+    const uploadUrls = await Promise.all(
+      file_uploads.map((file) => this.getFileUploadUrl(file)),
+    );
+
+    await Promise.all(
+      uploadUrls.map((uploadUrl, index) =>
+        this.uploadFile(uploadUrl.upload_url, file_uploads[index].file)
+      ),
+    );
+
+    return await Promise.all(
+      uploadUrls.map((uploadUrl, index) =>
+        this.completeFileUpload(uploadUrl.file_id, file_uploads[index])
+      ),
+    );
+  }
+
+  private async getFileUploadUrl(file: FileUploadV2) {
+    const fileMetaData = {
+      filename: file.filename,
+      length: file.length,
+      alt_text: file.alt_text,
+      snippet_type: file.snippet_type,
+    };
+    const response = await this.apiCall(
+      "files.getUploadURLExternal",
+      fileMetaData,
+    );
+
+    if (!response.ok) {
+      throw new Error(JSON.stringify(response.response_metadata));
+    }
+    return response;
+  }
+
+  private async completeFileUpload(fileID: string, file: FileUploadV2) {
+    const fileMetaData = {
+      files: JSON.stringify([{ id: fileID, title: file.title }]),
+      channel_id: file.channel_id,
+      initial_comment: file.initial_comment,
+      thread_ts: file.thread_ts,
+    };
+    const response = await this.apiCall(
+      "files.completeUploadExternal",
+      fileMetaData,
+    );
+    if (!response.ok) {
+      throw new Error(JSON.stringify(response.response_metadata));
+    }
+    return response;
+  }
+
+  private async uploadFile(
+    uploadUrl: string,
+    file: FileUploadV2["file"],
+  ) {
+    const response = await fetch(uploadUrl, {
+      headers: {
+        "Content-Type": typeof file === "string"
```

Comment:
I had overlooked how the node SDK behaves when a string is passed to the file field 😮. It's definitely not good for the behavior to differ between SDKs. I will make corrections so that the handling of the content field and file field behaves the same as in other SDKs.